alertmuting/model_alert_muting_rule_filter.go

```diff
@@ -9,9 +9,63 @@
 
 package alertmuting
 
+import (
+	"encoding/json"
+	"errors"
+)
+
 // Properties of an alert muting rule, in the form of a JSON object. **NOTE:** You can't create or update properties marked read-only. You receive read-only properties in response bodies for the following:    * **GET** `/alertmuting`   * **POST** `/alertmuting`   * **GET** `/alertmuting/{id}`   * **PUT** `/alertmuting/{id}`
 type AlertMutingRuleFilter struct {
-	NOT           bool   `json:"NOT,omitempty"`
-	Property      string `json:"property,omitempty"`
-	PropertyValue string `json:"propertyValue,omitempty"`
+	NOT           bool          `json:"NOT,omitempty"`
+	Property      string        `json:"property,omitempty"`
+	PropertyValue StringOrArray `json:"propertyValue,omitempty"`
+}
+
+type StringOrArray struct {
+	Values []string
+}
+
+func (soa *StringOrArray) MarshalJSON() ([]byte, error) {
+	if len(soa.Values) == 1 {
+		return json.Marshal(soa.Values[0])
+	} else if len(soa.Values) > 1 {
+		return json.Marshal(soa.Values)
+	}
+	return nil, errors.New("propertyValue has to be a string or array of strings")
+}
+
+func (soa *StringOrArray) UnmarshalJSON(b []byte) error {
+	if len(b) == 0 {
+		return nil // empty
+	}
+	// See if we can guess based on the first character
+	switch b[0] {
+	case '{':
```

Comment:
Wouldn't this be a JSON object in this case?  If so, how would it unmarshal into a string?